tests/pass-dep/libc/libc-eventfd.rs

```diff
@@ -0,0 +1,132 @@
+//@ignore-target-windows: No eventfd in windows
+//@ignore-target-apple: No eventfd in macos
+// test_race depends on a deterministic schedule.
+//@compile-flags: -Zmiri-preemption-rate=0
+
+use std::thread;
+
+fn main() {
+    test_read_write();
+    test_race();
+}
+
+fn test_read_write() {
+    let flags = libc::EFD_NONBLOCK | libc::EFD_CLOEXEC;
+    let fd = unsafe { libc::eventfd(0, flags) };
+    let sized_8_data: [u8; 8];
+    if cfg!(target_endian = "big") {
+        // Adjust the data based on the endianness of host system.
+        sized_8_data = 1_i64.to_be_bytes();
+    } else {
+        sized_8_data = 1_i64.to_le_bytes();
+    }
+    // Write 1 to the counter.
+    let res: i64 = unsafe {
+        libc::write(fd, sized_8_data.as_ptr() as *const libc::c_void, 8).try_into().unwrap()
+    };
+    assert_eq!(res, 8);
```

Comment:
Maybe create a write and read helper function in this file that wraps this logic for you